app/controllers/application_controller.rb

```diff
@@ -0,0 +1,17 @@
+class ApplicationController < ActionController::API
+  include ActiveSupport::Rescuable
+
+  rescue_from ActiveRecord::RecordInvalid do |e| render_error(e.to_s, :unprocessable_entity) end
+  rescue_from ActiveRecord::RecordNotFound do |e| render_error(e.to_s, :not_found) end
+  rescue_from AuthenticationService::Unauthorized do |e| render_error(e.to_s, :unauthorized) end
+
+  def generate_jwt(id, email)
+    JWT.encode({'id': id, 'email': email}, Rails.application.credentials.jwt_secret, 'HS256')
```

Comment:
```suggestion
    JWT.encode({'id': id, 'user': username}, Rails.application.credentials.jwt_secret, 'HS256')
```
username as the unique key - email is optional.

Reply:
this will be `'username' : username`